packages/kotti-ui/source/kotti-field-select/components/GenericSelectField.vue

```diff
@@ -0,0 +1,379 @@
+	.merge(Shared.isMultipleSchema)
+	.merge(Shared.isRemoteSchema)
+	.merge(Shared.isSingleSchema)
+	.omit({ value: true })
+	.extend({
+		helpTextSlot: z.array(z.unknown()).default(() => []),
+		isMultiple: z.boolean().default(false),
+		isRemote: z.boolean().default(false),
+		value: z.union([
+			Shared.isMultipleSchema.shape.value,
+			Shared.isSingleSchema.shape.value,
+		]),
+	})
+
+type SingleValue =
+	| KottiFieldSingleSelect.Value
+	| KottiFieldSingleSelectRemote.Value
+
+type MultiValue =
+	| KottiFieldMultiSelect.Value
+	| KottiFieldMultiSelectRemote.Value
+
+export default defineComponent<
+	KottiFieldSingleSelectRemote.PropsInternal & z.output<typeof propsSchema>
+>({
+	name: 'GenericSelectField',
+	components: {
+		ActionIcon,
+		FieldSelectOptions,
+		KtField,
+	},
+	props: makeProps(propsSchema),
+	setup(props, { emit: rawEmit }) {
+		const emit = (event: string, payload: unknown) =>
+			rawEmit('emit', { event, payload })
+
+		const field = useField<SingleValue | MultiValue>({
+			emit,
+			isEmpty: (value) =>
+				props.isMultiple
+					? (value as MultiValue).length === 0
+					: (value as SingleValue) === null,
+			props,
+			supports: KOTTI_FIELD_SELECT_SUPPORTS,
+		})
+
+		const inputRef = ref<HTMLInputElement | null>(null)
+
+		const localQuery = ref<string | null>(null)
+
+		const { isDropdownOpen, isDropdownMounted, ...selectTippy } =
+			useSelectTippy()
+
+		const selectedLabel = computed((): string | null => {
+			if (field.currentValue === null) return null
+			return (
+				props.options.find((option) => option.value === field.currentValue)
+					?.label ?? null
+			)
+		})
+
+		const { forceUpdateKey, forceUpdate } = useForceUpdate()
+
+		watch(isDropdownMounted, (isMounted) => {
+			if (isMounted) return
+
+			if (props.isRemote) {
+				if (props.query !== null) emit(UPDATE_QUERY, null)
+			} else localQuery.value = null
+		})
+
+		watch(isDropdownOpen, (isOpen) => {
+			if (!isOpen) return
+
+			inputRef.value?.focus()
+		})
+
+		const queryValue = computed(() =>
+			props.isRemote ? props.query : localQuery.value,
+		)
+
+		/**
+		 * keeps reference of already seen option labels so that
+		 * the tags can be rendered properly when filtering remotely
+		 */
+		const seenValueLabelMap = ref<Map<Shared.Value, string>>(new Map())
+
+		watch(
+			() => props.options,
+			() => {
+				seenValueLabelMap.value = new Map([
+					...seenValueLabelMap.value,
+					...props.options.map((option): [Shared.Value, string] => [
+						option.value,
+						option.label,
+					]),
+				])
+			},
+			{ immediate: true },
+		)
+
+		return {
+			collapsedTagCount: computed(() =>
+				props.isMultiple
+					? (field.currentValue as MultiValue).length - props.collapseTagsAfter
+					: 0,
+			),
+			field,
+			filteredOptions: computed(() =>
+				props.isRemote
+					? props.options
+					: props.options.filter(
+							(o) =>
+								localQuery.value === null ||
+								o.label.toLowerCase().includes(localQuery.value.toLowerCase()),
```

Comment:
apparently this is explicity requested by the user app even

Reply:
let's wait for future feature requests to decide on one of the following

1) adding an explicit attribute for searchValue - which is what the user can look for when querying
2) use one of those two "hacks" as a solution
  a) make the field "remote" and filter on the user app by more than just the `option.label` (aka by `option.label || option.value`
  b) make the `label` (since this is what the internal implementation uses know when the user queries) have the same value as the `value` and use the slot to show the label you want the user to see (e.g. { label: 'Label', value: 'Value' } --> becomes { label: 'Value', value: 'Value' } and the slot for option ACTUALLY shows 'Label'

Reply:
@FlorianWendelborn: consider fusejs.io to support fuzzy search